src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -35,10 +34,27 @@ export default function withAnnotations<P extends object>(
             this.setState({ ...this.state, annotation, action, error });
         };
 
+        handleActiveChange = (annotationId: string | null): void => {
+            this.setState({ ...this.state, activeAnnotationId: annotationId });
```

Comment:
You should be able to omit `...this.state` here. React will merge the state automatically.

Reply:
Interesting thing is, if I omit `...this.state`, eslint starts complaining about `no-unused-state` warning (https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unused-state.md)